File: Factory/FactoryMethod/Java/src/Pizza/NewYorkStyle/NewYorkStyleClamPizza.java
package eu.svetlin.design.patterns.factory.factorymethod.Pizza.NewYorkStyle;

import eu.svetlin.design.patterns.factory.factorymethod.Pizza.Pizza;

public class NewYorkStyleClamPizza extends Pizza {
    public NewYorkStyleClamPizza(){
        name = "New York Style Clam Pizza";
        dough = "Thin Crust Dough";
        sauce = "Marinara Sauce";

        toppings.add("Grated Reggiano Cheese");
        toppings.add("Fresh Clams from Long Island Sound");
    }
}

File: Factory/FactoryMethod/TypeScript/src/Pizza/NewYorkStyle/NewYorkStyleClamPizza.ts
import Pizza from '../Pizza';

class NewYorkStyleClamPizza extends Pizza {
  constructor() {
    super();
    this.name = 'New York Style Clam Pizza';
    this.dough = 'Thin Crust Dough';
    this.sauce = 'Marinara Sauce';

    this.toppings.push('Grated Reggiano Cheese');
    this.toppings.push('Fresh Clams from Long Island Sound');
  }
}

export default NewYorkStyleClamPizza;
